src/script.ts

```diff
@@ -5,12 +5,11 @@ import { optionalArgument, variadicArgument } from './util';
 
 export interface DagScriptExecuteOptions {
   inputs?: string[];
-  listInput?: string[];
+  args?: string[];
```

Comment:
there should be only one class for `ScriptExecuteOption`
keys, args, and inputs are supported on dag and non-dag execution

Reply:
`timeout` is a dag-only option, right?
I think I'll just move `keys` to `DagScriptExecuteOptions` and use `dagScriptExecuteFlatArgs` in `scriptExecuteFlatArgs`.